guides/micronaut-oracle-autonomous-db/micronaut-oracle-autonomous-db.adoc

```diff
@@ -125,44 +126,122 @@ source:controller/ThingController[]
+Create an `application-oraclecloud.yml` file in `src/main/resources` with this content:
+
+resource:application-oraclecloud.yml[]
 
-testResource:application-test.yml[]
+<1> Set the value of the `ocid` property with the database OCID unique identifier you saved when creating the database
+<2> Set the `walletPassword` property with a password to encrypt the wallet keys (must be at least 8 characters and include at least 1 letter and either 1 numeric or special character)
+<3> Set the `password` property with the `micronautdemo` schema user password you created
+<4> Change the profile name if you're not using the default, and optionally add a value for the path to the config file if necessary as described in the https://micronaut-projects.github.io/micronaut-oracle-cloud/latest/guide/#config-auth[Authentication section] of the Micronaut Oracle Cloud docs
 
-Create a `testcontainers.properties` file in `src/test/resources` with this content:
+=== Writing Tests
+
+Create a test to verify that database access works:
+
+test:ThingRepositoryTest[]
+
+== Testing the Application
+
+There are two options for running the tests; one is to run against the live database, and the other is to run tests locally with an Oracle database in a Docker container using https://www.testcontainers.org/[Testcontainers].
+
+To test using Testcontainers, create a `testcontainers.properties` file in `src/test/resources` with this content:
 
 testResource:testcontainers.properties[]
 
-Create a new Flyway migration SQL script in `src/main/resources/db/migration/V1__create-schema.sql` and add the following:
+and replace the generated `application-test.yml` with this:
 
-resource:db/migration/V1__create-schema.sql[]
+testResource:application-test.yml[tags=datasource|flyway]
 
-And finally, you can optionally edit `src/main/resources/logback.xml` and add the following to monitor the SQL queries that Micronaut Data performs:
+To test using the live database, replace the generated `application-test.yml` with this:
 
-[source,xml]
+testResource:application-test.yml[tags=flyway]
+
+Then configure your build to specify the `oraclecloud` environment so the datasource configured in `application-oraclecloud.yml` is used:
+
+:exclude-for-build:maven
+
+Edit `build.gradle` and add
+
+[source, groovy]
 ----
-<logger name='io.micronaut.data.query' level='debug' />
+test.systemProperty('micronaut.environments', 'oraclecloud')
 ----
 
-=== Writing Tests
+:exclude-for-build:
 
-Create a test to verify that database access works:
+:exclude-for-build:gradle
 
-test:ThingRepositoryTest[]
+Edit `pom.xml` and add
+
+[source, xml]
+----
+<plugin>
+    <groupId>org.apache.maven.plugins</groupId>
+    <artifactId>maven-surefire-plugin</artifactId>
+    <configuration>
+        <systemPropertyVariables>
+            <micronaut.environments>oraclecloud</micronaut.environments>
+        </systemPropertyVariables>
+    </configuration>
+</plugin>
+----
```

Comment:
This is overly complicated. You can just do:

```
@MicronautTest(environments=Environment.ORACLE_CLOUD)
```

To active the oracle cloud config for the test